pages/index.js

```diff
@@ -1,13 +1,26 @@
 import Head from "next/head";
 import RoastCard from "@/components/RoastCard";
 import { StyledList, StyledItem } from "@/lib/styled-components";
+import { useSession } from "next-auth/react";
 import useSWR from "swr";
 
 const fetcher = (...args) => fetch(...args).then((res) => res.json());
 
 export default function Home() {
+  const { data: session } = useSession();
   const { data, isLoading } = useSWR("api/roasts", fetcher);
-  if (isLoading) return <h1>... is Loading</h1>;
+  const { data: favorites, isLoading: favoritesLoading } = useSWR(
+    () => (session ? "api/user" : null),
```

Comment:
You probably don't need the function here.
```suggestion
    session ? "api/user" : null,
```

Reply:
Yes, this works as well. SWR docs just show it with the function so i thought it was needed

Reply:
Passing a function is for things that are computationally expensive, if I'm not mistaken. There could be things you need to calculate, which take a lot of time. For these calculations you could pass a function which gets executed only **once**. At least that's what I recall, I haven't checked the docs :P